frameworks/moveos-stdlib/src/natives/moveos_stdlib/json.rs

```diff
@@ -275,34 +290,358 @@ fn native_from_json(
+        let json_value = serialize_move_value_to_json(&field_layout.layout, value)?;
+        fields.insert(name.clone().into_string(), json_value);
+    }
+
+    Ok(JsonValue::Object(fields))
+}
+
+#[inline]
+fn native_to_json(
+    gas_params: &ToBytesGasParameters,
+    context: &mut NativeContext,
+    mut ty_args: Vec<Type>,
+    mut args: VecDeque<MoveValue>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert_eq!(ty_args.len(), 1);
+    debug_assert_eq!(args.len(), 1);
+
+    let mut cost = gas_params.base;
+
+    // pop type and value
+    let ref_to_val = pop_arg!(args, Reference);
+    let arg_type = ty_args.pop().unwrap();
+
+    // get type layout
+    let layout = match context.type_to_type_layout(&arg_type)? {
+        Some(layout) => layout,
+        None => {
+            return Ok(NativeResult::err(cost, E_JSON_SERIALIZATION_FAILURE));
+        }
+    };
+
+    let move_val = ref_to_val.read_ref()?.as_move_value(&layout);
+
+    let annotated_layout = match context.type_to_fully_annotated_layout(&arg_type)? {
+        Some(layout) => layout,
+        None => {
+            return Ok(NativeResult::err(cost, E_JSON_SERIALIZATION_FAILURE));
+        }
+    };
+
+    let annotated_move_val = move_val.decorate(&annotated_layout);
+
+    match serialize_move_value_to_json(&annotated_layout, &annotated_move_val) {
+        Ok(json_value) => {
+            let json_string = json_value.to_string();
+            cost += gas_params.per_byte_in_str * NumBytes::new(json_string.len() as u64);
+
+            Ok(NativeResult::ok(
+                cost,
+                smallvec![MoveValue::vector_u8(json_string.into_bytes())],
+            ))
+        }
+        Err(e) => {
+            debug!("Failed to serialize value: {:?}", e);
+
+            Ok(NativeResult::err(
+                cost,
+                STATUS_CODE_FAILED_TO_SERIALIZE_VALUE,
+            ))
+        }
+    }
+}
+
 /***************************************************************************************************
  * module
  **************************************************************************************************/
 
 #[derive(Debug, Clone)]
 pub struct GasParameters {
     pub from_bytes: FromBytesGasParameters,
+    pub to_bytes: ToBytesGasParameters,
 }
 
 impl GasParameters {
     pub fn zeros() -> Self {
         Self {
             from_bytes: FromBytesGasParameters::zeros(),
+            to_bytes: ToBytesGasParameters::zeros(),
         }
     }
 }
 
 pub fn make_all(gas_params: GasParameters) -> impl Iterator<Item = (String, NativeFunction)> {
-    let natives = [(
-        "native_from_json",
-        make_native(gas_params.from_bytes, native_from_json),
-    )];
+    let natives = [
+        (
+            "native_from_json",
+            make_native(gas_params.from_bytes, native_from_json),
+        ),
+        (
+            "native_to_json",
+            make_native(gas_params.to_bytes, native_to_json),
+        ),
```

Comment:
The new native function gas_params needs to be optional, like https://github.com/rooch-network/rooch/blob/a3ae131ea83a9da6bce3d8461e2a95860f581f5c/frameworks/rooch-framework/src/natives/rooch_framework/bitcoin_address.rs#L247-L255

Reply:
OK, I'll fix it.